src/twisted/trial/_synctest.py

```diff
@@ -1022,15 +1023,25 @@ def shortDescription(self):
         return desc
 
 
-    def getSkip(self):
+    def getSkip(self) -> Tuple[bool, Optional[str]]:
         """
         Return the skip reason set on this test, if any is set. Checks on the
         instance first, then the class, then the module, then packages. As
-        soon as it finds something with a C{skip} attribute, returns that.
-        Returns L{None} if it cannot find anything. See L{TestCase} docstring
-        for more details.
-        """
-        return util.acquireAttribute(self._parents, 'skip', None)
+        soon as it finds something with a C{skip} attribute, returns that in
+        a tuple (L{True}, L{str}).
+        If the C{skip} attribute does not exist, look for C{__unittest_skip__}
+        and C{__unittest_skip_why__} attributes which are set by the standard
+        library L{unittest.skip} function.
+        Returns (L{False}, L{None}) if it cannot find anything.
+        See L{TestCase} docstring for more details.
+        """
+        skipReason = util.acquireAttribute(self._parents, 'skip', None)
+        doSkip = skipReason is not None
+        if skipReason is None:
+            doSkip = getattr(self, "__unittest_skip__", False)
+            if doSkip:
+                skipReason = getattr(self, "__unittest_skip_why__", "")
+        return (doSkip, skipReason)
```

Comment:
Can we have something like this and not change the return value for getSkip  ?

It will no longer return `None` ...but `False` instead, but I think that this can reduce downstream changes. 

```suggestion
        return skipReason or doSkip
```

Now, this is only testing infrastructure public dode... so if something breaks, it will break at the testing phase and not in production.

So feel free to merge as it is.

i am very happy with the new automated tests :)